VitruvioHost/Plugins/Vitruvio/Source/VitruvioEditor/Private/VitruvioComponentDetails.cpp

```diff
@@ -311,10 +331,56 @@ void AddArrayWidget(const TArray<TSharedRef<IDetailTreeNode>> DetailTreeNodes, I
 			FDetailWidgetRow ArrayDefaultWidgetsRow;
 			TSharedPtr<SWidget> ArrayNameWidget;
 			TSharedPtr<SWidget> ArrayValueWidget;
-	
+
 			DetailPropertyRow->GetDefaultWidgets(ArrayNameWidget, ArrayValueWidget, ArrayDefaultWidgetsRow, true);
 			ValueRow.NameContent()[ArrayNameWidget.ToSharedRef()];
-			ValueRow.ValueContent()[ArrayValueWidget.ToSharedRef()];
+
+			if (UFloatArrayAttribute* FloatArrayAttribute = Cast<UFloatArrayAttribute>(Attribute))
+			{
+				if (FloatArrayAttribute->GetEnumAnnotation() && FloatArrayAttribute->Values.Num() > 0)
+				{
+					ValueRow.ValueContent()[CreateArrayEnumWidget<UFloatArrayAttribute, double, UFloatEnumAnnotation>(
+										FloatArrayAttribute, FloatArrayAttribute->GetEnumAnnotation(), DetailPropertyRow->GetPropertyHandle())
+										.ToSharedRef()];
+				}
+				else
+				{
+					ValueRow.ValueContent()[ArrayValueWidget.ToSharedRef()];
+				}
+			}
+			else if (UStringArrayAttribute* StringArrayAttribute = Cast<UStringArrayAttribute>(Attribute))
+			{
+				if (StringArrayAttribute->GetEnumAnnotation() && StringArrayAttribute->GetEnumAnnotation()->Values.Num() > 0)
+				{
+					ValueRow.ValueContent()[CreateArrayEnumWidget<UStringArrayAttribute, FString, UStringEnumAnnotation>(
+										StringArrayAttribute, StringArrayAttribute->GetEnumAnnotation(), DetailPropertyRow->GetPropertyHandle())
+										.ToSharedRef()];
+				}
+				else if (StringArrayAttribute->GetColorAnnotation())
+				{
+					auto ColorStringProperty = DetailPropertyRow->GetPropertyHandle();
+					auto ColorSetter = [ColorStringProperty](FLinearColor NewColor) {
+						ColorStringProperty->SetValue(TEXT("#") + NewColor.ToFColor(true).ToHex());
+					};
+					auto ColorGetter = [ColorStringProperty]()
+					{
+						FString Value;
+						ColorStringProperty->GetValue(Value);
+						return FLinearColor(FColor::FromHex(Value));
+					};
+					
+					ValueRow.ValueContent()[CreateColorInputWidget(ColorSetter, ColorGetter).ToSharedRef()];
+				}
+				else
+				{
+					ValueRow.ValueContent()[ArrayValueWidget.ToSharedRef()];
+				}
+			}
+			else
+			{
+				ValueRow.ValueContent()[ArrayValueWidget.ToSharedRef()];
+			}
```

Comment:
Maybe it would make sense to change this code to:
```
TSharedRef<SWidget> ValueContent;
..... (ifs set ValueContent value)
ValueRow.ValueContent()[ValueContent ? ValueContent : ArrayValueWidget.ToSharedRef()];
```